common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/messaging/repository/DiscoveryApiRepositoryInformation.java

```diff
@@ -81,7 +85,7 @@ private DiscoveryApiResponse getDiscoverApiResponse() throws IOException, Interr
         throwExceptionOnUnexpectedStatusCode(response.statusCode(), SC_OK);
 
         DiscoveryApiResponse discoveryApiResponse = objectMapper.readValue(response.body(), DiscoveryApiResponse.class);
-        log.atTrace().log("Discovery API response: {}", discoveryApiResponse);
+        log.trace("Discovery API response: {}", discoveryApiResponse);
```

Comment:
I meant to ask, is this API only invoked once at the startup to determine the version, or is it invoked N times? If it's the latter, then we should - at least eventually - plan caching this IMO.

Reply:
Thanks for catching the logger part.
About caching - `ApplicationInfoProvider` class which is used across all modules as `User-Agent` header value provider will call `RepositoryInformation::getRepositoryVersion` only once and keep it in class member field.
If this solution is somehow faulty, we can plan some caching, but for now, I don't see a reason to do so.